depends: update xcb-util packages to latest versions

fanquake opened the pull request:
Update libxcb and related package versions:

libXau 1.0.9 -> 1.0.12
libxcb 1.14 -> 1.17
libxcb_util 0.4.0 -> 0.4.1
libxcb_util_cursor 0.1.5 -> 0.1.6
libxcb_util_image 0.4.0 -> 0.4.1
libxcb_util_keysyms 0.4.0 -> 0.4.1
libxcb_util_render 0.3.9 -> 0.3.10
libxcb_util_wm 0.4.1 -> 0.4.2
xcb_proto 1.15.2 -> 1.17.0

These recently became static in #33537.

Also update expat 2.4.8 -> 2.7.3.

fanquake committed:
depends: libXau 1.0.12

fanquake committed:
depends: libxcb-util 0.4.1

fanquake committed:
depends: libxcb-util-image 0.4.1

fanquake committed:
depends: libxcb-util-keysyms 0.4.1

fanquake committed:
depends: libxcb-util-renderutil 0.3.10

fanquake committed:
depends: libxcb-util-wm 0.4.2

hebasto commented:
cc @laanwj

fanquake committed:
depends: libxcb 1.15

fanquake committed:
depends: libxcb-util-cursor 0.1.6

fanquake commented:
Guix Build (x86_64):
```bash
3a70d632167f9e57328d9b8373a5eb9936428148feddf5bc1f9834f412ea4248  guix-build-daafaef87fee/output/aarch64-linux-gnu/SHA256SUMS.part
ca1c7ad00632d0ee3fdfad498a7c6cb5246240a9e49c89bee508355ee7dbb832  guix-build-daafaef87fee/output/aarch64-linux-gnu/bitcoin-daafaef87fee-aarch64-linux-gnu-debug.tar.gz
a5a5c5a22159a912f573915727b5a4cdbb1b270fdf099f8b465c3ef940496e20  guix-build-daafaef87fee/output/aarch64-linux-gnu/bitcoin-daafaef87fee-aarch64-linux-gnu.tar.gz
a660c559c1f2acac2bb5dbd1a4f6e147cd3d3119971f33451080d7b6c0e7f01e  guix-build-daafaef87fee/output/arm-linux-gnueabihf/SHA256SUMS.part
22c17f6290c1106761fe0abdb2d6f0cb7b00e4825f6e4640883641a33f52a283  guix-build-daafaef87fee/output/arm-linux-gnueabihf/bitcoin-daafaef87fee-arm-linux-gnueabihf-debug.tar.gz
f968d81191379c0f709ea30f658e123470188eb0b9c128376de2de89b6ef58b3  guix-build-daafaef87fee/output/arm-linux-gnueabihf/bitcoin-daafaef87fee-arm-linux-gnueabihf.tar.gz
01a83422e6769ac7783ed1e7f67603f639f897ae19f8de67e1dc544a56264024  guix-build-daafaef87fee/output/arm64-apple-darwin/SHA256SUMS.part
e51cdb5b264f22481afd9d2805d4ff4c56b864a9fe5e17420f4737ccf3d661b3  guix-build-daafaef87fee/output/arm64-apple-darwin/bitcoin-daafaef87fee-arm64-apple-darwin-codesigning.tar.gz
d4358129da3279a29cbcf190d332e432852caa88efa50d12c42c61d493977eb5  guix-build-daafaef87fee/output/arm64-apple-darwin/bitcoin-daafaef87fee-arm64-apple-darwin-unsigned.tar.gz
205ca8b27940e63142989eec61701d6ab8b59d6f8b5a63466c25504e92a5241d  guix-build-daafaef87fee/output/arm64-apple-darwin/bitcoin-daafaef87fee-arm64-apple-darwin-unsigned.zip
78d9fa5b08627ffc8199d6115d483f8f96b4235066abd058d53b92a9866952d1  guix-build-daafaef87fee/output/dist-archive/bitcoin-daafaef87fee.tar.gz
64709c3c7ac5190a6f77c53b03a4487ef3b42fc6302b17dcaeafd09e48252c68  guix-build-daafaef87fee/output/powerpc64-linux-gnu/SHA256SUMS.part
a1f504e950d8dbd3365ab7e00061bedb6fe4c3477da9146942bf4ce860c11db3  guix-build-daafaef87fee/output/powerpc64-linux-gnu/bitcoin-daafaef87fee-powerpc64-linux-gnu-debug.tar.gz
30d94f319a2467a8c0c3bf09746504e2246c615b673870d33e97355bd5211278  guix-build-daafaef87fee/output/powerpc64-linux-gnu/bitcoin-daafaef87fee-powerpc64-linux-gnu.tar.gz
c02f44ca0276e9fa85aaf56b6c2be717512de4ea7c1024f410c0ca9da58cc126  guix-build-daafaef87fee/output/riscv64-linux-gnu/SHA256SUMS.part
385870441e711f32cd20c4877d1b3d1a949003f2189007acd5e55681d06ebdb2  guix-build-daafaef87fee/output/riscv64-linux-gnu/bitcoin-daafaef87fee-riscv64-linux-gnu-debug.tar.gz
18ef362f5072b8c361cfa6401b5fd2d9e89436cbd032f43e1ed0b13f70e82642  guix-build-daafaef87fee/output/riscv64-linux-gnu/bitcoin-daafaef87fee-riscv64-linux-gnu.tar.gz
a9148de45a7d106a8d8cf2a1b4fa3578afeecf63ba873bdfeb7c0ffe91c53562  guix-build-daafaef87fee/output/x86_64-apple-darwin/SHA256SUMS.part
8df6dc51196646b59d02a28d888cd608f202d5a5dd724f56cf33e5c89c21f81c  guix-build-daafaef87fee/output/x86_64-apple-darwin/bitcoin-daafaef87fee-x86_64-apple-darwin-codesigning.tar.gz
65adc7eb17328ba3bfa398839e8e367d4cd0c42514665846659c14f8b05f3158  guix-build-daafaef87fee/output/x86_64-apple-darwin/bitcoin-daafaef87fee-x86_64-apple-darwin-unsigned.tar.gz
1332fd93c4e7db9edd93b327e3ad4b9ef77194a62bed8a445f25ffb6c0bd0d02  guix-build-daafaef87fee/output/x86_64-apple-darwin/bitcoin-daafaef87fee-x86_64-apple-darwin-unsigned.zip
9d7b15bcb1b064057b2d4b6e1e1b42783507eeca752c63c4af98e0ccd4211d65  guix-build-daafaef87fee/output/x86_64-linux-gnu/SHA256SUMS.part
c29bcad502f86e792a2a0941df182d00fd7f5cca1afbf4fce773e6c6f8eec964  guix-build-daafaef87fee/output/x86_64-linux-gnu/bitcoin-daafaef87fee-x86_64-linux-gnu-debug.tar.gz
b3b88aa88d042d5bbfd1bff05dd42c59b56df079d32314be32e60ee25d4cf507  guix-build-daafaef87fee/output/x86_64-linux-gnu/bitcoin-daafaef87fee-x86_64-linux-gnu.tar.gz
a1206cf0707dfb23b9a011fb070ecc1c6ba8f54963118013895f4a263e0ff152  guix-build-daafaef87fee/output/x86_64-w64-mingw32/SHA256SUMS.part
ac6e7df1e0f6ef3320e49de9b0bb2d6ac18d3f73b5364fd9ef9a734d9e661245  guix-build-daafaef87fee/output/x86_64-w64-mingw32/bitcoin-daafaef87fee-win64-codesigning.tar.gz
145575be6b9f87a246b146d67feab0bc4f07c765b4ac3b61665e771fe368c2d8  guix-build-daafaef87fee/output/x86_64-w64-mingw32/bitcoin-daafaef87fee-win64-debug.zip
87e10e19e6b8d52b995624c570ff64414e1f99bf7873077613beac2e42017ebb  guix-build-daafaef87fee/output/x86_64-w64-mingw32/bitcoin-daafaef87fee-win64-setup-unsigned.exe
dee2952de01e0bfef07d1cc5081815832ba8b120209478951cacf8782b41d15d  guix-build-daafaef87fee/output/x86_64-w64-mingw32/bitcoin-daafaef87fee-win64-unsigned.zip
```

hebasto reviewed (commented):
My Guix build:
```
aarch64
3a70d632167f9e57328d9b8373a5eb9936428148feddf5bc1f9834f412ea4248  guix-build-daafaef87fee/output/aarch64-linux-gnu/SHA256SUMS.part
ca1c7ad00632d0ee3fdfad498a7c6cb5246240a9e49c89bee508355ee7dbb832  guix-build-daafaef87fee/output/aarch64-linux-gnu/bitcoin-daafaef87fee-aarch64-linux-gnu-debug.tar.gz
a5a5c5a22159a912f573915727b5a4cdbb1b270fdf099f8b465c3ef940496e20  guix-build-daafaef87fee/output/aarch64-linux-gnu/bitcoin-daafaef87fee-aarch64-linux-gnu.tar.gz
a660c559c1f2acac2bb5dbd1a4f6e147cd3d3119971f33451080d7b6c0e7f01e  guix-build-daafaef87fee/output/arm-linux-gnueabihf/SHA256SUMS.part
22c17f6290c1106761fe0abdb2d6f0cb7b00e4825f6e4640883641a33f52a283  guix-build-daafaef87fee/output/arm-linux-gnueabihf/bitcoin-daafaef87fee-arm-linux-gnueabihf-debug.tar.gz
f968d81191379c0f709ea30f658e123470188eb0b9c128376de2de89b6ef58b3  guix-build-daafaef87fee/output/arm-linux-gnueabihf/bitcoin-daafaef87fee-arm-linux-gnueabihf.tar.gz
01a83422e6769ac7783ed1e7f67603f639f897ae19f8de67e1dc544a56264024  guix-build-daafaef87fee/output/arm64-apple-darwin/SHA256SUMS.part
e51cdb5b264f22481afd9d2805d4ff4c56b864a9fe5e17420f4737ccf3d661b3  guix-build-daafaef87fee/output/arm64-apple-darwin/bitcoin-daafaef87fee-arm64-apple-darwin-codesigning.tar.gz
d4358129da3279a29cbcf190d332e432852caa88efa50d12c42c61d493977eb5  guix-build-daafaef87fee/output/arm64-apple-darwin/bitcoin-daafaef87fee-arm64-apple-darwin-unsigned.tar.gz
205ca8b27940e63142989eec61701d6ab8b59d6f8b5a63466c25504e92a5241d  guix-build-daafaef87fee/output/arm64-apple-darwin/bitcoin-daafaef87fee-arm64-apple-darwin-unsigned.zip
78d9fa5b08627ffc8199d6115d483f8f96b4235066abd058d53b92a9866952d1  guix-build-daafaef87fee/output/dist-archive/bitcoin-daafaef87fee.tar.gz
64709c3c7ac5190a6f77c53b03a4487ef3b42fc6302b17dcaeafd09e48252c68  guix-build-daafaef87fee/output/powerpc64-linux-gnu/SHA256SUMS.part
a1f504e950d8dbd3365ab7e00061bedb6fe4c3477da9146942bf4ce860c11db3  guix-build-daafaef87fee/output/powerpc64-linux-gnu/bitcoin-daafaef87fee-powerpc64-linux-gnu-debug.tar.gz
30d94f319a2467a8c0c3bf09746504e2246c615b673870d33e97355bd5211278  guix-build-daafaef87fee/output/powerpc64-linux-gnu/bitcoin-daafaef87fee-powerpc64-linux-gnu.tar.gz
c02f44ca0276e9fa85aaf56b6c2be717512de4ea7c1024f410c0ca9da58cc126  guix-build-daafaef87fee/output/riscv64-linux-gnu/SHA256SUMS.part
385870441e711f32cd20c4877d1b3d1a949003f2189007acd5e55681d06ebdb2  guix-build-daafaef87fee/output/riscv64-linux-gnu/bitcoin-daafaef87fee-riscv64-linux-gnu-debug.tar.gz
18ef362f5072b8c361cfa6401b5fd2d9e89436cbd032f43e1ed0b13f70e82642  guix-build-daafaef87fee/output/riscv64-linux-gnu/bitcoin-daafaef87fee-riscv64-linux-gnu.tar.gz
a9148de45a7d106a8d8cf2a1b4fa3578afeecf63ba873bdfeb7c0ffe91c53562  guix-build-daafaef87fee/output/x86_64-apple-darwin/SHA256SUMS.part
8df6dc51196646b59d02a28d888cd608f202d5a5dd724f56cf33e5c89c21f81c  guix-build-daafaef87fee/output/x86_64-apple-darwin/bitcoin-daafaef87fee-x86_64-apple-darwin-codesigning.tar.gz
65adc7eb17328ba3bfa398839e8e367d4cd0c42514665846659c14f8b05f3158  guix-build-daafaef87fee/output/x86_64-apple-darwin/bitcoin-daafaef87fee-x86_64-apple-darwin-unsigned.tar.gz
1332fd93c4e7db9edd93b327e3ad4b9ef77194a62bed8a445f25ffb6c0bd0d02  guix-build-daafaef87fee/output/x86_64-apple-darwin/bitcoin-daafaef87fee-x86_64-apple-darwin-unsigned.zip
9d7b15bcb1b064057b2d4b6e1e1b42783507eeca752c63c4af98e0ccd4211d65  guix-build-daafaef87fee/output/x86_64-linux-gnu/SHA256SUMS.part
c29bcad502f86e792a2a0941df182d00fd7f5cca1afbf4fce773e6c6f8eec964  guix-build-daafaef87fee/output/x86_64-linux-gnu/bitcoin-daafaef87fee-x86_64-linux-gnu-debug.tar.gz
b3b88aa88d042d5bbfd1bff05dd42c59b56df079d32314be32e60ee25d4cf507  guix-build-daafaef87fee/output/x86_64-linux-gnu/bitcoin-daafaef87fee-x86_64-linux-gnu.tar.gz
a1206cf0707dfb23b9a011fb070ecc1c6ba8f54963118013895f4a263e0ff152  guix-build-daafaef87fee/output/x86_64-w64-mingw32/SHA256SUMS.part
ac6e7df1e0f6ef3320e49de9b0bb2d6ac18d3f73b5364fd9ef9a734d9e661245  guix-build-daafaef87fee/output/x86_64-w64-mingw32/bitcoin-daafaef87fee-win64-codesigning.tar.gz
145575be6b9f87a246b146d67feab0bc4f07c765b4ac3b61665e771fe368c2d8  guix-build-daafaef87fee/output/x86_64-w64-mingw32/bitcoin-daafaef87fee-win64-debug.zip
87e10e19e6b8d52b995624c570ff64414e1f99bf7873077613beac2e42017ebb  guix-build-daafaef87fee/output/x86_64-w64-mingw32/bitcoin-daafaef87fee-win64-setup-unsigned.exe
dee2952de01e0bfef07d1cc5081815832ba8b120209478951cacf8782b41d15d  guix-build-daafaef87fee/output/x86_64-w64-mingw32/bitcoin-daafaef87fee-win64-unsigned.zip
```

hebasto reviewed (commented):
Maybe update `xcb_proto` as well?

hebasto committed:
depends: libxcb 1.17.0

--without-launchd was removed in
https://gitlab.freedesktop.org/xorg/lib/libxcb/-/commit/ccdef1a8a5b6956a3fb7115d634db213433dca75.

hebasto committed:
depends: xcb_proto 1.17.0

fanquake committed:
depends: expat 2.7.3

Remove the patch, as expat has bumped it's minimum to 3.15.
Remove the lto related workaround. I haven't seen issues with CMakes
TestBigEndian.

hebasto reviewed (commented):
My Guix build:
```
aarch64
17cd36aa7eaae8cbe242efa364e55699b1568a6d0cb2c2e1ac82781b46a5f174  guix-build-f541b92cf2bb/output/aarch64-linux-gnu/SHA256SUMS.part
69985b56e1c6ff521f4f49ba102cb53275106dfd076be98361068ef113379fa3  guix-build-f541b92cf2bb/output/aarch64-linux-gnu/bitcoin-f541b92cf2bb-aarch64-linux-gnu-debug.tar.gz
4feb585aaf30ecc8b8c1c1fa32e4a8fa9e7f3e67286825e006807f714f15b039  guix-build-f541b92cf2bb/output/aarch64-linux-gnu/bitcoin-f541b92cf2bb-aarch64-linux-gnu.tar.gz
43b3f41f64a01fb29b7ffbde60f038c47bb70a231dfea6ad9ae2cc332a07d84b  guix-build-f541b92cf2bb/output/arm-linux-gnueabihf/SHA256SUMS.part
9627c13bd778709e84ac6eb1ea0efaa7708cbc3f06296ec63decb9d20bea5c7e  guix-build-f541b92cf2bb/output/arm-linux-gnueabihf/bitcoin-f541b92cf2bb-arm-linux-gnueabihf-debug.tar.gz
ef8a0e054d85dbbc2cbb2b9b9e81f75b31c19aeb4925376d2e585a44f93f7d5e  guix-build-f541b92cf2bb/output/arm-linux-gnueabihf/bitcoin-f541b92cf2bb-arm-linux-gnueabihf.tar.gz
0da56cd5bc6152d8bf7b47237bd345dfeaf869dcd029202357e58fa392c119ce  guix-build-f541b92cf2bb/output/arm64-apple-darwin/SHA256SUMS.part
2d12d14ae2c3ec8b57fc04fecd9d387378c7e99b91f1dd18bcdb3f36b72db5df  guix-build-f541b92cf2bb/output/arm64-apple-darwin/bitcoin-f541b92cf2bb-arm64-apple-darwin-codesigning.tar.gz
e73fcd6631cbedeb63a17877cbad68dda690b90ab52a65196772ae6aed156bce  guix-build-f541b92cf2bb/output/arm64-apple-darwin/bitcoin-f541b92cf2bb-arm64-apple-darwin-unsigned.tar.gz
e4f8a0666a4ab61758d8bc74bd30bd03eaea347d96fcdd917bb07d6cf6228a2e  guix-build-f541b92cf2bb/output/arm64-apple-darwin/bitcoin-f541b92cf2bb-arm64-apple-darwin-unsigned.zip
bf0213008981b9766d41fca3083bd38390bbf931f2f8abb9545082178cddfe67  guix-build-f541b92cf2bb/output/dist-archive/bitcoin-f541b92cf2bb.tar.gz
f72f0a6d6f84e188c506aa66a552a152d21a5442fc14645fdafbd4a1eccb1ed9  guix-build-f541b92cf2bb/output/powerpc64-linux-gnu/SHA256SUMS.part
f0f238caa907216e3ed4a7552c180d9e8345b089a131473986f825a92b902178  guix-build-f541b92cf2bb/output/powerpc64-linux-gnu/bitcoin-f541b92cf2bb-powerpc64-linux-gnu-debug.tar.gz
2145729d6239063579ab9a3615ba19dd3e4ea3e595539098b59b66ad38fa59ce  guix-build-f541b92cf2bb/output/powerpc64-linux-gnu/bitcoin-f541b92cf2bb-powerpc64-linux-gnu.tar.gz
6f8e952b2a1a345ba752564e3a6624e6b5b49a3187c95fd324a6c8d86792267a  guix-build-f541b92cf2bb/output/riscv64-linux-gnu/SHA256SUMS.part
f9e2f35018bc736fada3cb86055b79b9a8c118a45f3c15a0e4173c01f7df3ff2  guix-build-f541b92cf2bb/output/riscv64-linux-gnu/bitcoin-f541b92cf2bb-riscv64-linux-gnu-debug.tar.gz
335f3899502cf2f7e45d4ed6ca6ba9bf215442ec5f8d9681e36cc6b561a91f9b  guix-build-f541b92cf2bb/output/riscv64-linux-gnu/bitcoin-f541b92cf2bb-riscv64-linux-gnu.tar.gz
16d735eea70875c733c0e89a9292d5221b365d509e90ce45a3952d591d77233a  guix-build-f541b92cf2bb/output/x86_64-apple-darwin/SHA256SUMS.part
f60d9486a339da0ec32bd6e73e67ec070d907114d4c7afef302d6d3268572299  guix-build-f541b92cf2bb/output/x86_64-apple-darwin/bitcoin-f541b92cf2bb-x86_64-apple-darwin-codesigning.tar.gz
75be31e777c86407971e8091d9631ece5aa499280d433aaf66d5c220a48c2a5d  guix-build-f541b92cf2bb/output/x86_64-apple-darwin/bitcoin-f541b92cf2bb-x86_64-apple-darwin-unsigned.tar.gz
18a7ac66da36d309b32214b2f6e00992c82c5c16cc03179ca2ddf2d83119787a  guix-build-f541b92cf2bb/output/x86_64-apple-darwin/bitcoin-f541b92cf2bb-x86_64-apple-darwin-unsigned.zip
f4592af2eb751cdc387c139726df24a614a352bd9d354d576e843bb4442a5084  guix-build-f541b92cf2bb/output/x86_64-linux-gnu/SHA256SUMS.part
eb6d8266819d31f2883aa92fef019046b207bd947e216d697752227a29f47c77  guix-build-f541b92cf2bb/output/x86_64-linux-gnu/bitcoin-f541b92cf2bb-x86_64-linux-gnu-debug.tar.gz
6a6eafbeae04c189586fd426f457d9096b78b12a33eb9a210904bd5a165de43a  guix-build-f541b92cf2bb/output/x86_64-linux-gnu/bitcoin-f541b92cf2bb-x86_64-linux-gnu.tar.gz
5e354d72ad7f47ea6e2e3409d5d9d7a32887499ec0782a6deda7dc3970f864bf  guix-build-f541b92cf2bb/output/x86_64-w64-mingw32/SHA256SUMS.part
69f2ae08b5e008039b7f257c7de8802f711aea58dbdcc360a2e7382965b60c5e  guix-build-f541b92cf2bb/output/x86_64-w64-mingw32/bitcoin-f541b92cf2bb-win64-codesigning.tar.gz
922681601699f83a52d8e4f7ef98683bb19de40a2c381a0aa12760c6d08b74c5  guix-build-f541b92cf2bb/output/x86_64-w64-mingw32/bitcoin-f541b92cf2bb-win64-debug.zip
be9050362c576ee8700727e696939621128817c4c09253e2d76fb03e252b685a  guix-build-f541b92cf2bb/output/x86_64-w64-mingw32/bitcoin-f541b92cf2bb-win64-setup-unsigned.exe
e980b90f95c7b0b2aefd276f54d4b5a1edd2556da8969a223e976fd1b8689e28  guix-build-f541b92cf2bb/output/x86_64-w64-mingw32/bitcoin-f541b92cf2bb-win64-unsigned.zip
```

hebasto reviewed (approved):
ACK f541b92cf2bb7f8f708f147bd320becf48587d14, due to the last commit, I also tested an LTO build on Ubuntu 25.10 using GCC 15.2.0.
